test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/ExpressionRewriterTest.cs

```diff
@@ -19,6 +19,41 @@ namespace Microsoft.AspNet.Mvc.Razor.Compilation
     public class ExpressionRewriterTest
     {
         [Fact]
+        public void ExpressionRewriter_DoesNotThrowsOnUnknownTypes()
+        {
+            // Arrange
+            var source = @"
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Razor;
+
+public class ExamplePage : RazorPage
+{
+    public IViewComponentHelper Component { get; set; }
+
+    public override async Task ExecuteAsync()
+    {
+        Write(
+            await Component.InvokeAsync(
+                ""SomeComponent"",
+                item => new HelperResult((__razor_template_writer) => WriteLiteralTo(__razor_template_writer, ""Hello World""))));
+        }
+    }
+";
+            var tree = CSharpSyntaxTree.ParseText(source);
+
+            // Allow errors here because of an anomaly where Roslyn (depending on code sample) will finish compilation
+            // without diagnostic errors. This test case replicates that scenario by allowing a semantic model with
+            // errors to be visited by the expression rewriter to validate unexpected exceptions aren't thrown.
+            var compilation = Compile(tree, allowErrors: true);
+            var semanticModel = compilation.GetSemanticModel(tree, ignoreAccessibility: true);
+            var rewriter = new ExpressionRewriter(semanticModel);
+
+            // Act - This should not throw, if the type cannot be resolved for the lambda it should no-op.
```

Comment:
Might be improved slightly if it's possible to assert that the rewriter no-oped. Would
``` c#
// still in Arrange
var root = tree.GetRoot();

// Act
var result = rewriter.Visit(root);

// Assert
Assert.Equal(root, result);
```

Put another way, do Linq expressions nodes provide useful `Equals()` implementations? If not is there a solid `SyntaxNodeComparator` available?